docs/source/guides/coding_style.rst

```diff
@@ -0,0 +1,225 @@
+
+In fact, these validations should not be done inside. Experiment is correct to assume that when it requires an instance and a solution, it should get one as arguments. If the arguments were optional, then they should appear with a default value in the method. In practice, we *want* Experiment to fail when there are incorrect (unexpected) arguments.
+
+The actual validations should be done outside (before) the creation of the Experiment so as to be sure to provide Experiment always with an instance and a solution.
+
+These kinds of validations can be done in javascript (python) with the use of TypeScript (python type-hints). While TypeScript is not yet present in Cornflow-app, python type-hints are very well present in cornflow-server and cornflow-client.
+
+In this case, the correct code would be the same but without ifs:
+
+.. code-block:: javascript
+
+    export class Experiment extends ExperimentCore {
+      static MyInstance = Instance
+      static MySolution = Solution
+
+      constructor (instance, solution) {
+        super(instance, solution)
+
+        this.colors = this.solution.colors
+
+        this.sizeCustomerPoint = 10
+        this.sizeBasePoint = 30
+        this.sizeSourcePoint = 20
+          this.offsetCustomers = this.instance.data.sources.length + 1
+        this.colorBase = '#65A0D1'
+        this.colorSource = '#D35A5A'
+
+        this.horizon = this.instance.data.parameters.horizon
+        this.unit = this.instance.data.parameters.unit
+      }
+
+
+Indentation
+---------------
+
+original code::
+
+.. code-block:: python
+
+    driver = shift["driver"]
+    time_cost = self.instance.get_driver_property(driver, "TimeCost")
+    layover_cost = self.instance.get_driver_property(driver, "LayoverCost")
+
+    trailer = shift["trailer"]
+    dist_cost = self.instance.get_trailer_property(trailer, "DistanceCost")
+
+    current_stop = None
+    layover = False
+
+
+    for stop in shift["route"]:
+        previous_stop = current_stop
+        current_stop = stop["location"]
+        if stop["layover_before"] == 1:
+            layover = True
+        if previous_stop is not None:
+            distance = self.instance.get_distance_between(
+                previous_stop, current_stop
+            )
+            time = self.instance.get_time_between(previous_stop, current_stop)
+            if self.is_source(current_stop):
+                time += self.instance.get_source_property(
+                    current_stop, "setupTime"
+                )
+            elif self.is_customer(current_stop):
+                time += self.instance.get_customer_property(
+                    current_stop, "setupTime"
+                )
+            total_cost += distance * dist_cost + time * time_cost
+
+
+    if layover:
+        total_cost += layover_cost
+
+In general, I do not like excessive indentation. One way to do that with an if inside a for is with something like this::
```

Comment:
I would avoid to have I in the documentation. Rephrase to something like:

"In general, excessive indentation can make the code hard to review and understand. One way..."